src/commands/chat/chat.ts

```diff
@@ -1,48 +1,85 @@
-import { CommandInteraction, SlashCommandBuilder } from'discord.js';
+import { CommandInteraction, SlashCommandBuilder } from 'discord.js';
+import fs from 'fs'
 
+// ** this is only for attempting to read the file that already exists
+const pathToPersonalityJson = '../../../config/personality.json'
+// ** see below for a remark on this. ~Line 63 - 1/20/2024 JOSEPH
+
+// TODO: Comment | What does this mean?
 module.exports = {
   data: new SlashCommandBuilder()
     .setName("chat")
     .setDescription("Sends a message to ChatGPT")
     .addStringOption((option) => {
       return option
-              .setName("prompt")
-              .setDescription("What do you want ChatGPT to do?")
-              .setRequired(true)
+        .setName("prompt")
+        .setDescription("What do you want ChatGPT to do?")
+        .setRequired(true)
     })
     .setDMPermission(true),
-            
-            
+
+  // TODO: Comment | When is this executed? Why is it significant?
   async execute(interaction: CommandInteraction) {
+
     // trigger a deferred response, otherwise the 3-second timeout will kill this request
     await interaction.deferReply();
 
+    // TODO: Comment
     const OpenAI = require('openai');
 
+    // TODO: Comment
     const openai = new OpenAI({
       apiKey: process.env.OPENAI_API_KEY,
     });
 
-            
+    // TODO: Comment
     console.debug(`Discord interaction options: ${JSON.stringify(interaction.options)}`);
 
-    // @ts-ignore
+    // TODO: Comment
+    // @ts-ignore  // THIS IS TERRIBLE THAT THIS MUST BE DONE -JOSEPH
     console.debug(`User submitted prompt: ${interaction.options.getString("prompt")}`);
-  
-    const response = await openai.chat.completions.create({
-      model: process.env.MODEL ?? "gpt-3.5-turbo",
-      messages: [
 
-        // TODO: Read these in from a JSON file configurable by the user
-        {role: "system", content: "You are a sassy and sarcastic assistant, but you don't need to tell me how sassy and sarcastic you are - just be it"},
-        {role: "system", content: "You are a gamer and reference dank memes often that you found on your favorite subreddit"},
-        {role: "system", content: "You should limit your responses to 2000 characters"},
+    // this will be populated with the personality prompts as well as the users prompt,
+    // it will then be passed in where appropriate below. 
+    let prompts = [];
+    // attempt to open personality json file / create empty default if isnt there
+    try {
+      // access
+      const personalityJson = require(pathToPersonalityJson);
+      // scrape contained prompts
+      for (let i = 0; i < personalityJson.prompts.length; i++) {
```

Comment:
I would use the `Array.prototype.forEach()` here instead of writing out a for loop:

`personalityJson.prompts.forEach((prompt) => prompts.push({ role: "system", content: prompt })`